src/bio.c

```diff
@@ -200,24 +224,23 @@ void *bioProcessBackgroundJobs(void *arg) {
         listNode *ln;
 
         /* The loop always starts with the lock hold. */
-        if (listLength(bio_jobs[type]) == 0) {
-            pthread_cond_wait(&bio_newjob_cond[type],&bio_mutex[type]);
+        if (listLength(bio_jobs[worker]) == 0) {
+            pthread_cond_wait(&bio_newjob_cond[worker], &bio_mutex[worker]);
             continue;
         }
         /* Pop the job from the queue. */
```

Comment:
Perhaps we can fix the outdated comment(changed in https://github.com/redis/redis/commit/1317b7c2338777343841e8ea3a83c120bac1747e) in the meantime.
```suggestion
        /* Get the first job from the queue. */
```